celer/tests/test_lasso.py

```diff
@@ -218,6 +218,27 @@ def test_weights_lasso():
     np.testing.assert_raises(ValueError, clf1.fit, X=X, y=y)
 
 
+def test_infinite_weights():
+    np.random.seed(1)
+    n_samples, n_features = 50, 100
+    X = np.random.randn(n_samples, n_features)
+    y = np.random.randn(n_samples)
+
+    weights = np.ones(n_features)
+    weights[0] = np.inf
+    alpha_max = norm(X.T @ y / weights, ord=np.inf) / n_samples
+
+    # assert convergence
+    with warnings.catch_warnings():
```

Comment:
this is overcomplicated. Increase the number of iterations in order to make the warning disappear instead

in general: fix the source of the warning instead of silencing it